src/unit/test_ziplist.c

```diff
@@ -0,0 +1,1049 @@
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistGetElementOutOfRangeReverse(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    zl = createList(); /* "hello", "foo", "quux", "1024" */
+    p = ziplistIndex(zl, -5);
+    TEST_ASSERT(p == NULL);
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistIterateThroughFullList(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    zl = createList();
+    p = ziplistIndex(zl, 0);
+    while (ziplistGet(p, &entry, &elen, &value)) {
+        TEST_ASSERT(p != NULL);
+        p = ziplistNext(zl, p);
+    }
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistIterateThroughListFrom1ToEnd(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    zl = createList();
+    p = ziplistIndex(zl, 1);
+    while (ziplistGet(p, &entry, &elen, &value)) {
+        TEST_ASSERT(p != NULL);
+        p = ziplistNext(zl, p);
+    }
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistIterateThroughListFrom2ToEnd(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    zl = createList();
+    p = ziplistIndex(zl, 2);
+    while (ziplistGet(p, &entry, &elen, &value)) {
+        TEST_ASSERT(p != NULL);
+        p = ziplistNext(zl, p);
+    }
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistIterateThroughStartOutOfRange(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    zl = createList();
+    p = ziplistIndex(zl, 4);
+    TEST_ASSERT(p == NULL);
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistIterateBackToFront(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    zl = createList();
+    p = ziplistIndex(zl, -1);
+    while (ziplistGet(p, &entry, &elen, &value)) {
+        TEST_ASSERT(p != NULL);
+        p = ziplistPrev(zl, p);
+    }
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistIterateBackToFrontDeletingAllItems(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    zl = createList();
+    p = ziplistIndex(zl, -1);
+    while (ziplistGet(p, &entry, &elen, &value)) {
+        TEST_ASSERT(p != NULL);
+        zl = ziplistDelete(zl, &p);
+        p = ziplistPrev(zl, p);
+    }
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistDeleteInclusiveRange0To0(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    zl = createList(); /* "hello", "foo", "quux", "1024" */
+
+    p = ziplistIndex(zl, 0);
+    TEST_ASSERT(ziplistCompare(p, (unsigned char *)"hello", 5));
+    int orig_len = ziplistLen(zl);
+
+    zl = ziplistDeleteRange(zl, 0, 1);
+
+    TEST_ASSERT(ziplistCompare(p, (unsigned char *)"foo", 3));
```

Comment:
I believe p is getting invalidated here because of the ziplistDeleteRange. I think you need to reinitialize it.

Reply:
Looks like Ping may have beat me to it with #537. Sorry about that.

Reply:
No worries I tried to run the daily tests before merging it, and there was another known issue so I merged it anyways.  :/